src/lat/lattice-functions.h

```diff
@@ -306,6 +306,21 @@ void ComposeCompactLatticeDeterministic(
     fst::DeterministicOnDemandFst<fst::StdArc>* det_fst,
     CompactLattice* composed_clat);
 
+/// This function returns true if the CompactLattice clat has unique
+/// left phone context of length n, and false if it does not.
+bool HasUniquePhoneContext(const CompactLattice &clat,
+       const TransitionModel &tmodel, int32 n);
```

Comment:
Delete them.  If we need them for some reason later on, we can reconstruct
them from the history.
Dan



On Wed, Jan 6, 2016 at 2:22 PM, Hossein Hadian <notifications@github.com>
wrote:

> In src/lat/lattice-functions.h
> <https://github.com/kaldi-asr/kaldi/pull/414#discussion_r49018092>:
>
> > @@ -306,6 +306,21 @@ void ComposeCompactLatticeDeterministic(
> >      fst::DeterministicOnDemandFst<fst::StdArc>* det_fst,
> >      CompactLattice* composed_clat);
> >
> > +/// This function returns true if the CompactLattice clat has unique
> > +/// left phone context of length n, and false if it does not.
> > +bool HasUniquePhoneContext(const CompactLattice &clat,
> > +       const TransitionModel &tmodel, int32 n);
>
> Actually I just checked what composition does, not how it does it. I will
> spend some time studying it.
> So now that we don't need phone expansion, should I leave the functions
> be, or delete them ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/414/files#r49018092>.
>